### PR TITLE
IA-1645 Don't leak db connection in unit tests

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/IA-1645

My approach here was to remove `DbSingleton` and use `BeforeAndAfterAll` to init/close the slick database between tests. This is how slick recommends to do it. I also made it so we don't have to run liquibase every time.

It seems to fix the issue for me, and it doesn't seem to add too much time to the test run.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
@Qi77Qi fyi, this is working great for me

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/TestComponent.scala [hunk 1]
```diff
@@ -2,23 +2,60 @@ package org.broadinstitute.dsde.workbench.leonardo.db
...
+  override def afterAll(): Unit = {
+    dbRef.close()
+    super.afterAll()
+  }
+
+  // This is a bit duplicative of DbReference.init but that method returns a Resource[F, DbRef[F]]
+  // which doesn't play nicely with ScalaTest BeforeAndAfterAll. This version returns an F[DbRef[F]].
+  private def initDbRef: IO[DbRef[IO]] =
+    for {
+      concurrentPermits <- Semaphore[IO](100)
```
Comment (kyuksel) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/TestComponent.scala]: Should `100` come from a config?

**Review by kyuksel — APPROVED** [review]
Verified `http/test` succeeds when run within an `sbt` shell 👍 
One minor comment inline.
Thanks a lot for fixing this 🙏

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/TestComponent.scala (hunk 1 above)
Comment (rtitle) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/TestComponent.scala]: Ah yeah good catch, this is already in reference.conf. Will update.

**Comment by Qi77Qi** [comment]
🎉 👍 thanks for fixing this!

**Pull request merged**